lib/Socket.js

```diff
@@ -61,7 +60,8 @@ class Socket extends EventEmitter {
       args = args || {};
       let address = args.address || 'localhost';
 
-      if (!url.parse(address).port) {
+      // If no port was provided, add the default port.
+      if (!/(?::\d{2,4})/.test(address)) {
```

Comment:
Surprisingly, this was a necessary change to make these tests work. The previous code didn't work at all and would result in `address` being `localhost:5555:4444`, because this is how `url.parse` handles the types of strings that we were providing it here:

![image](https://cloud.githubusercontent.com/assets/873012/25458201/a65f2294-2a9e-11e7-87f5-c783d58d38fa.png)

Over time we can work on more robust input validation with support for `ws://` urls and the like.